### ISSUE TITLE
Image srcset is wrong if image meta size is in string

### ISSUE THREAD

**Issue by Shebo** [deliciousbrains/wp-amazon-s3-and-cloudfront#409]
Hi there,
I came across an annoying-but-simple bug I want to share with you.

## TLDR
`find_image_size_from_width` condition is ignoring matched `$size['width']` when it's a string.
https://github.com/deliciousbrains/wp-amazon-s3-and-cloudfront/blob/761e98dad46938e58cf36a85fb536f146a7fe9ea/classes/as3cf-plugin-compatibility.php#L792

## Explanation
My WP site holds many images, and their respective metadata (`_wp_attachment_metadata` meta_key) sizes are inserted to the DB as strings instead of integers. I guess that it's because my site is 6 years old (been upgraded of course).

The function `find_image_size_from_width` that in `as3cf-plugin-compatibility.php` is comparing the sizes (which i couldn't understand why, looks like wp already doing this before the filter).

In any case, `find_image_size_from_width` is checking that the requested filename and size are legit, but my old images sizes (`$size['width']`) are inserted in the serialized array as strings and not as integers.
Because the comparison done with triple equal sign (`===`), the string type doesn't match the int type and the function returns `null`.
https://github.com/deliciousbrains/wp-amazon-s3-and-cloudfront/blob/761e98dad46938e58cf36a85fb536f146a7fe9ea/classes/as3cf-plugin-compatibility.php#L792

I can override this with some filtering, but my suggestion is to replace the triple equal sign (`===`) with a double (`==`) so it won't compare the variable type, or convert the `$size['width']` to an integer before the condition.

**Comment by ianmjones** [comment]
Thanks for the bug report @Shebo.

**Comment by ianmjones** [comment]
Fixed in [release 1.4](https://github.com/deliciousbrains/wp-amazon-s3-and-cloudfront/releases/tag/1.4)
